packages/grpc-native-core/ext/channel_credentials.cc

```diff
@@ -148,9 +187,42 @@ NAN_METHOD(ChannelCredentials::CreateSsl) {
         "createSsl's second and third arguments must be"
         " provided or omitted together");
   }
+
+  verify_peer_options verify_options = {NULL, NULL, NULL};
+  if (info.Length() >= 4 && info[3]->IsObject()) {
```

Comment:
I don't think the length check is necessary here. As far as I can tell, indexing past the end returns undefined. In addition, I would like to also have an error thrown if the argument is provided but not the correct type, like with the other arguments.